projects/storefrontlib/src/cms-components/checkout/components/payment-method/payment-form/payment-form.component.ts

```diff
@@ -128,11 +128,22 @@ export class PaymentFormComponent implements OnInit, OnDestroy {
     this.shippingAddress$ = this.checkoutDeliveryService.getDeliveryAddress();
     this.loading$ = this.checkoutPaymentService.getSetPaymentDetailsResultProcess();
 
-    this.checkboxSub = this.showSameAsShippingAddressCheckbox().subscribe(
-      (shouldShowCheckbox: boolean) => {
-        // this operation makes sure the checkbox is not checked if not shown and vice versa
+    this.showSameAsShippingAddressCheckbox$ = combineLatest([
+      this.countries$,
+      this.shippingAddress$,
+    ]).pipe(
+      map(([countries, address]) => {
+        return (
+          address?.country &&
+          !!countries.filter(
+            (country: Country): boolean =>
+              country.isocode === address.country.isocode
+          ).length
+        );
+      }),
+      tap((shouldShowCheckbox) => {
         this.sameAsShippingAddress = shouldShowCheckbox;
```

Comment:
Since we have showSameAsShippingAddressCheckbox$ as an observable, we don't really need this static prop and `tap`, I guess. IMO can be postponed to a refactor ticket. (not MUST for 2.0; refactor for 3.0)

Reply:
Definitely, but usage in a lot of places make that hard at the moment.